### PR TITLE
HPCC-12683 Use interleaved memory policy

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
Looks like it still needs rebasing

File: roxie/roxiemem/roxiemem.cpp [hunk 1]
```diff
@@ -166,6 +170,28 @@ static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages,
 #else
     heapUseHugePages = false;
 
+#ifdef _USE_NUMA
+    int has_numa = numa_available() + 1;
+    if (has_numa)
```
Comment (richardkchapman) [inline roxie/roxiemem/roxiemem.cpp]: Personally I'd probably code it as 
  bool has_numa = numa_available() != -1;
  if (has_numa)
  ...

just seems clearer ...

Comment (richardkchapman) [inline roxie/roxiemem/roxiemem.cpp]: (actually I might use >= 0 - it's not clear if other -ve values are ever returned)

**Comment by mckellyln** [comment]
Incorrectly rebased to wrong branch and could not recover cleanly.  
New PR issued with https://github.com/hpcc-systems/HPCC-Platform/pull/6798

**Pull request closed without merging**